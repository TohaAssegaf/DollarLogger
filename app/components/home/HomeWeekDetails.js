import styles from './styles'
import HomeHeader from './HomeHeader'
import PaymentContributionList from '~/app/components/payment/PaymentContributionList'
import * as PaymentUtils from '~/app/lib/PaymentUtils'
import React from 'react'
import { View } from 'react-native'
import { withNavigation } from 'react-navigation';
import { connect } from 'react-redux'

class HomeWeekDetails extends React.Component {
  navigateToUpdatePayment(paymentId: number) {
    this.props.navigation.navigate(
      Routes.UPDATE_PAYMENT,
      { payment: this.props.payments.find(payment => payment.id == paymentId) })
  }

  render() {
    return (
      <View style={styles.screen}>
        <HomeHeader />
        <PaymentContributionList
          onTapCell={
            paymentContribution => this.navigateToUpdatePayment(paymentContribution.paymentId)}
          paymentContributions={this.props.paymentContributions}
        />
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    payments: PaymentUtils.filterCurrentWeekPayments(state.payment.payments),
    paymentContributions:
      PaymentUtils.filterCurrentWeekPaymentContributions(state.payment.payments),
  }
}

export default withNavigation(connect(mapStateToProps)(HomeWeekDetails))
