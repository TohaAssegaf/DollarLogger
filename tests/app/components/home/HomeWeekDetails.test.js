import HomeWeekDetails from '~/app/components/home/HomeWeekDetails'
import MockDate from 'mockdate'
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

it('renders correctly', () => {
  const date = new Date(2018, 3, 8)
  MockDate.set(date)
  const store = mockStore({
    budget: { total: 25000 },
    payment: { payments: [
      {
        id: 1,
        total: 1,
        name: "Test payment",
        date,
        paymentContributions: [
          {
            total: 1,
            date,
            displayName: "Test payment",
            paymentId: 1,
          },
        ],
      }
    ]}})
  const rendered = renderer.create(
    <Provider store={store}><HomeWeekDetails date={date} /></Provider>).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('does not include previous week payments', () => {
  const oldDate = new Date(2018, 3, 8)
  const currentDate = new Date(2018, 3, 9)
  MockDate.set(currentDate)
  const store = mockStore({
    budget: { total: 25000 },
    payment: { payments: [
      {
        id: 1,
        total: 1,
        name: "Test payment",
        date: oldDate,
        paymentContributions: [
          {
            total: 1,
            date: oldDate,
            displayName: "Test payment",
            paymentId: 1,
          },
        ],
      }
    ]}})
  const rendered = renderer.create(
    <Provider store={store}><HomeWeekDetails date={currentDate} /></Provider>).toJSON();
  expect(rendered).toMatchSnapshot();
});
