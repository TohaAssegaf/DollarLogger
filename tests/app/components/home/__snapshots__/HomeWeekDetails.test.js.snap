// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not include previous week payments 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#222222",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 32,
          }
        }
      >
        $250.00
      </Text>
      <Text>
        Remaining this week
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 32,
          }
        }
      >
        $0.00
      </Text>
      <Text>
        Spent so far
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#222222",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 32,
          }
        }
      >
        $249.99
      </Text>
      <Text>
        Remaining this week
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 32,
          }
        }
      >
        $0.01
      </Text>
      <Text>
        Spent so far
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "date": 2018-04-08T04:00:00.000Z,
          "displayName": "Test payment",
          "paymentId": 1,
          "total": 1,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "#222222",
                "borderBottomWidth": 0.5,
                "flexDirection": "row",
                "height": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 0.15,
                  "paddingHorizontal": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                4/8
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 0.6,
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                Test payment
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 0.25,
                  "paddingRight": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                $0.01
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
